lib/doorkeeper/oauth/pre_authorization.rb

```diff
@@ -97,7 +97,14 @@ def validate_resource_owner_authorize_for_client
       end
 
       def validate_redirect_uri
-        return false if redirect_uri.blank?
+        # 4.1.1.  Authorization Request
+        #   redirect_uri
+        #     OPTIONAL.  As described in Section 3.1.2.
+        #
+        # @see https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.1
+        return true if redirect_uri.blank? &&
+                       client.redirect_uri.present? &&
```

Comment:
I'm checking for the presence of `client.redirect_uri` because I think it's possible for a Doorkeeper application to not have a pre-registered redirect URI. My understanding is that if an application allows the authorization code flow and/or implicit flow and that application doesn't have a pre-registered redirect URI then that is a misconfiguration but, since I think it is possible, I decided to handle that situation by rejecting authorization requests that don't include a `redirect_uri` param if the target application does not have a pre-registered redirect URI.